.github/workflows/publish-dist.yml

```diff
@@ -30,37 +30,22 @@ jobs:
           set -ex
           SHA=$(cat SHA)
 
-          # multiple parallel workflows race against each other, so the final
-          # `git push` may fail
-          rc=1
-          for retry in $(seq 5); do
-              git clone https://github.com/${{ github.repository }}-dist.git dist-repo
-              rm -rf dist-repo/dist
-              cp -r dist/ package-lock.json dist-repo/
+          git clone https://github.com/${{ github.repository }}-dist.git dist-repo
```

Comment:
I think there should be a `git checkout --orphan` here somewhere, to make it super clear that we are producing commits without history.

Reply:
Sure, I can add a `git checkout --orphan floating-in-space` or so here.

Reply:
In the `node_modules` stuff I just create an empty repo, init, add the files, commit, tag, and push from there.  I think that would be easier here, too.

Reply:
... or use what @allisonkarlitskaya is proposing below. That also sounds quite interesting, just want to make sure that it works as intended.

Reply:
Moved to Lis'es proposal, which makes this even more clear -- it always starts with an empty repo.

Reply:
*thumps-up*